### PR TITLE
Optional sudo usage (#36)

### PR DESCRIPTION
**Pull request by DavidGOrtega**
- https://github.com/iterative/setup-cml/pull/36

### CODE REVIEW

**Comment by DavidGOrtega** [comment]
@casperdcl I cant now merge without at least one approval

**Review by casperdcl — APPROVED** [review]

File: action.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ inputs:
     description: The version of CML to install (e.g. '3.0.0').
     default: latest
     required: false
+  sudo:
+    desciption: A flag to disable the use sudo while installing CML.
```
Comment (casperdcl) [inline action.yml:10]:
```suggestion
    desciption: A flag to disable the use of sudo whilst installing CML.
```

**Review by casperdcl — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -49,6 +49,20 @@ steps:
       version: '3.0.0'
 ```
 
+Self-Hosted example:
```
Comment (casperdcl) [inline README.md:52]:
```suggestion
Self-hosted example:
```

**Review by casperdcl — APPROVED** [review]
I presume there's no problem with prefixing whitespace (`exec(' echo hello')`)

**Comment by 0x2b3bfa0** [comment]
Failing tests because we depend on [vega-lite](https://www.npmjs.com/package/vega-lite) and it's trying to use a nullish coalescing operator `??` on Node JS 12, but it wasn't supported until 14. Related to https://github.com/vega/vega-lite/pull/6959?

**Comment by 0x2b3bfa0** [comment]
See also [vega/vega-lite#v5.0.0](https://github.com/vega/vega-lite/releases/tag/v5.0.0) release notes: _no more ES5 builds._

**Comment by 0x2b3bfa0** [comment]
We're installing all the `vega` packages without any version pinning, and with methods other than `package-lock.json` 🙀 

https://github.com/iterative/setup-cml/blob/fbe960885f26b2965cd88995323ad83da1e662c8/src/utils.js#L35-L37

**Comment by 0x2b3bfa0** [comment]
Approving and merging because checks are [failing on the main branch](https://github.com/iterative/setup-cml/runs/3528795329?check_suite_focus=true#step:13:13) anyway. Thanks to #37 scheduled checks we should notice each time something like this happens.

**Review by 0x2b3bfa0 — APPROVED** [review]

**Pull request merged**